lua/pixelui/core/cl_color.lua

```diff
@@ -130,4 +130,32 @@ function PIXEL.GetRainbowColor()
     lastCol = HSVToColor((time * 50) % 360, 1, 1)
 
     return lastCol
+end
+
+local colorMeta = FindMetaTable("Color")
+
+colorMeta.Copy = PIXEL.CopyColor
+
+function colorMeta:Offset(offset)
+    self.r = self.r + offset
+    self.g = self.g + offset
+    self.b = self.b + offset
+    return self
+end
+
+function colorMeta:Lerp(t, to)
+    self.r = lerp(t, self.r, to.r)
+    self.g = lerp(t, self.g, to.g)
+    self.b = lerp(t, self.b, to.b)
+    self.a = lerp(t, self.a, to.a)
+    return self
+end
+
+function colorMeta:IsLight()
+    local _, _, light = PIXEL.ColorToHSL(self)
```

Comment:
Might be a bit more helpful to be calling that var "lightness".
Also, localise ColorToHSL.